settings.gradle.kts

```
@@ -1,15 +1,13 @@
 plugins {
-    id("com.gradle.enterprise") version "3.17"
+    id("com.gradle.develocity") version "3.17"
     id("org.gradle.toolchains.foojay-resolver-convention") version "0.8.0"
 }
 
-val isCiBuild = System.getenv("CI") != null
-
-gradleEnterprise {
+develocity {
     buildScan {
-        termsOfServiceUrl = "https://gradle.com/terms-of-service"
-        termsOfServiceAgree = "yes"
-        publishAlwaysIf(isCiBuild)
+        termsOfUseUrl = "https://gradle.com/help/legal-terms-of-use"
+        termsOfUseAgree = "yes"
+        publishing.onlyIf { System.getenv("CI") != null }
```

```suggestion
        uploadInBackground = false
        publishing.onlyIf { System.getenv("CI") != null }
```

I think we need this based on:
https://docs.gradle.com/enterprise/gradle-plugin/current/#configuring_background_uploading

And I think this is not related to the migration, it should have been on before too (same docs in legacy too). @szpak did you ever have a problem where a scan wasn't available?

---

I don't recall that situation, however, I don't often use build scans, so I might miss it.

---

Maybe because there is some time consumed do save local cache. Probably we should set it to `false`, as you suggested.